### DISCUSSION TITLE
MPC in concrete

### DISCUSSION THREAD

**Discussion by bimintelligence** [do-mpc/do-mpc#353] [General]
Hi everyone, I want to use mpc in concrete with water pipe.
Temperatrue control with cool water, is it possible to achieve use do-mpc

**Comment by sergiolucia** [comment]
If you can model your system as a dynamic system (see the examples) and define a control task (using a cost function) with constraints, you will be able to use do-mpc for it. For some background information you can also check our webpage [www.do-mpc.com](https://www.do-mpc.com/)
